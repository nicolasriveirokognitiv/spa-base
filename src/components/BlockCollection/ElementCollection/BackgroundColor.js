import React from 'react';
import useTestMode from 'Hooks/useTestMode';
import { toolPropTypes } from 'Constants/propTypes';
import { Input } from 'antd';
import { elementCategories } from 'Constants/global';

function BackgroundColor({ onChange, value, propertyKey, defaultValue, testId }) {
    const testProps = useTestMode(testId);
    const handleChange = e => onChange(e.target.value, propertyKey);

    return (
        <Input
            value={value}
            placeholder={defaultValue}
            defaultValue={defaultValue}
            onChange={handleChange}
            {...testProps}
        />
    );
}

BackgroundColor.propTypes = toolPropTypes;
BackgroundColor.defaultProps = {
    category         : elementCategories.STYLE,
    propertyKey      : 'backgroundColor',
    label            : 'Background color',
    wrapperClassName : 'type-color-picker',
    defaultValue     : '#ffc900',
    onChange         : () => {},
};

export default BackgroundColor;
