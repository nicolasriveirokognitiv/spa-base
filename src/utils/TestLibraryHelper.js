/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import initStore from 'State/store';

const store = initStore();

const renderWithRedux = children => render(
    <Provider store={store}>
        {children}
    </Provider>,
);

export * from '@testing-library/react';
export { renderWithRedux, render };
