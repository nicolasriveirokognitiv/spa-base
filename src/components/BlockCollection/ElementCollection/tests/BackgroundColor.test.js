import React from 'react';
import { render, fireEvent } from 'Utils/TestLibraryHelper';
import EditorContext from 'State/editorContext';
import BackgroundColor from '../BackgroundColor';

describe('BackgroundColor', () => {
    let container;
    let props;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
        props = {
            onChange : jest.fn(),
            value    : 'chunk',
        };
    });

    const renderBackgroundColor = p => render(
        <EditorContext.Provider value={{ isTestModeActive: true }}>
            <BackgroundColor {...props} {...p} />
        </EditorContext.Provider>,
    );

    test('Should render Correctly', () => {
        expect(<BackgroundColor />).toBeDefined();
    });

    test('Should match snapshot', async () => {
        const { asFragment } = render(<BackgroundColor />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Input text should be the same in value', async () => {
        const { getByTestId } = renderBackgroundColor({ testId: 'background-color-input' });
        const input = getByTestId('background-color-input');
        expect(input.value).toBe('chunk');
    });

    test('should call onChange prop on changes in input', () => {
        const { getByTestId } = renderBackgroundColor({ testId: 'background-color-input' });
        fireEvent.change(getByTestId('background-color-input'), { target: { value: 'foo' } });
        expect(props.onChange).toHaveBeenCalledWith('foo', 'backgroundColor');
    });
});
