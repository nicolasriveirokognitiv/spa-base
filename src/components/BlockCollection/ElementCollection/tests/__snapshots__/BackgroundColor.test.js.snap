// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundColor Should match snapshot 1`] = `
<DocumentFragment>
  <input
    class="ant-input"
    placeholder="#ffc900"
    type="text"
    value="#ffc900"
  />
</DocumentFragment>
`;
